gpii/node_modules/flowManager/test/SolutionsRegistryDataSourceTests.js

```diff
@@ -0,0 +1,320 @@
+                    args: [gpii.tests.solutionsRegistry.requestOptions]
+                }, {
+                    event: "{solutionsRegistryMockRepoLoad}.events.onRead",
+                    listener: "gpii.tests.solutionsRegistry.checkNamedRegistry",
+                    args: [
+                        "Expecting named solutions registry from repository",
+                        jqUnit.assertDeepEq,
+                        gpii.tests.solutionsRegistry.repositoryDarwinSolutions,
+                        "{arguments}.0"
+                    ]
+                },
+                { funcName: "fluid.log", args: ["Solutions Registry Mock Repo Load, getting named repository registry -- END"] },
+                { funcName: "fluid.log", args: ["Solutions Registry Mock Repo Load, getting named local registry -- START"] },
+                {
+                    funcName: "{solutionsRegistryMockRepoLoad}.get",
+                    args: [{os: gpii.tests.solutionsRegistry.win32PlatformId}]
+                }, {
+                    event: "{solutionsRegistryMockRepoLoad}.events.onRead",
+                    listener: "gpii.tests.solutionsRegistry.checkOneRegistry",
+                    args: [
+                        "Expecting named solutions registries, but from the local file system",
+                        gpii.tests.solutionsRegistry.win32PlatformId,
+                        "{arguments}.0"
+                    ]
+                },
+                { funcName: "fluid.log", args: ["Solutions Registry Mock Repo Load, getting all registries -- START"] }
+            ]
+        }]
+    }]
+});
+
+// Test environment for LFM solutions registry data source where solutions are
+// fetched from the repository
+fluid.defaults("gpii.tests.solutionsRegistry.dataSource.local.mockRepository.env", {
+    gradeNames: ["fluid.test.testEnvironment"],
+    components: {
+        tester: {
+            type: "gpii.tests.solutionsRegistry.local.mockRepository.testCaseHolder"
+        }
+    }
+});
+kettle.test.bootstrap("gpii.tests.solutionsRegistry.dataSource.local.mockRepository.env");
+
+// Test component for the LFM solutions registry data source where it fails to
+// load from the source code repository.
+fluid.defaults("gpii.tests.solutionsRegistry.dataSource.local.failLoadFromRepository", {
+    gradeNames: ["gpii.flowManager.solutionsRegistry.dataSource.local"],
+    components: {
+        revisionRequester: {
+            type: "gpii.flowmanager.revisionRequester",
+            options: {
+                cloudURL: "http://gpii.net"
+            }
+        }
+    },
+    listeners: {
+        "loadSolutions.loadFromLocalDisk": {
+            listener: "{that}.loadFromLocalDisk",
+            priority: "first"
+        },
+        "loadSolutions.getRevision": {
+            listener: "{revisionRequester}.getRevision",
+            priority: "after:loadFromLocalDisk"
+        },
+        "loadSolutions.loadFromRepository": {
```

Comment:
Same as a previous comment. This listener should be the only one kept for overriding the corresponding listener in the base component. Other 3 that are identical with the base component can be removed from here.

Reply:
Right.